### ISSUE TITLE
Error: Uncaught (in promise): DataCloneError

### ISSUE THREAD

**Issue by ShadowQueen** [VadimDez/ng2-pdf-viewer#159]
Hi,
I keep getting the following error:

"Error: Uncaught (in promise): DataCloneError
   at resolvePromise (http://localhost:52504/node_modules/zone.js/dist/zone.js:665:25)
   at Anonymous function (http://localhost:52504/node_modules/zone.js/dist/zone.js:716:17)
   at ZoneDelegate.prototype.invokeTask (http://localhost:52504/node_modules/zone.js/dist/zone.js:365:13)
   at onInvokeTask (eval code:3971:25)
   at ZoneDelegate.prototype.invokeTask (http://localhost:52504/node_modules/zone.js/dist/zone.js:365:13)
   at Zone.prototype.runTask (http://localhost:52504/node_modules/zone.js/dist/zone.js:166:21)
   at drainMicroTaskQueue (http://localhost:52504/node_modules/zone.js/dist/zone.js:546:25)
   at invoke (http://localhost:52504/node_modules/zone.js/dist/zone.js:424:25)"

Everything seems to be OK If I use [src] = "'./pdf-test.pdf'", but if I try to read from web API for example 
[src] = "'http://localhost:42000/api/Documents/GetPath/del__8430'"   - error appears.
Basically I want to read from response, I don't now is this possible?
Any suggestions ?

Thank you for your time .

**Comment by Andikki** [comment]
I get this error whenever the source url is faulty in some way (for example, returns 404 Not Found)
> Error: Uncaught (in promise): DataCloneError: Failed to execute 'postMessage' on 'Worker'

**Comment by ShadowQueen** [comment]
Hi,
When I try to use URL directly everything is ok (just call API in new tab), I get a possibility for downloading the document (I can not preview directly). I think URL is not faulty.

**Comment by ShadowQueen** [comment]
Maybe this will help. I will share an image. Any suggestion? 
![error](https://user-images.githubusercontent.com/16406109/30531316-c0d8151a-9c4d-11e7-9936-361f96080fe0.png)

**Comment by ShadowQueen** [comment]
Please close this issue, I found the problem. I was blind.
Just need to set  withCredentials: true, works like a charm.

**Comment by Andikki** [comment]
Still the component shouldn't behave like this on invalid sources.

**Comment by ShadowQueen** [comment]
I agree with you, the error is little misleading. But author explained in the readme.MD file, we should use Options object for loading protected PDF.
{
 url: 'https://vadimdez.github.io/ng2-pdf-viewer/pdf-test.pdf',
 withCredentials: true
}
This trick resolves my issue.

**Comment by N1MA** [comment]
I get the same error when Internet Download Manager(IDM) is enabled on chrome. IDM prompts to download the pdf and pdf viewer throws this error

**Comment by Anujmoglix** [comment]
@ShadowQueen  can you write full code bez it is vary confusing from Readme.md to get the information .

**Comment by Anujmoglix** [comment]
It is working with that trick but issue is that pdf is not scrolling vertically :(
